modu/source_repository/source_repository.py

```diff
@@ -36,8 +37,10 @@ class SourceRepository:
     """
 
     def __init__(self, src_root: Path):
+        sys.setrecursionlimit(1500)
```

Comment:
Let's leave this alone, because I think this will effect all recursion in the program and the default should be more than enough for levels of recursion we want to perform when excluding directories.